react/features/base/react/components/web/MultiSelectAutocomplete.js

```diff
@@ -0,0 +1,316 @@
+        this.setState({
+            filterValue,
+            isOpen: Boolean(this.state.items.length)
+                && Boolean(filterValue),
+            items: filterValue ? this.state.items : []
+        }, this._sendQuery);
+    }
+
+    /**
+     * Retries the query on retry.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onRetry() {
+        this._sendQuery();
+    }
+
+    /**
+     * Sends a query to the resourceClient.
+     *
+     * @returns {void}
+     */
+    _sendQuery() {
+        const text = this.state.filterValue;
+
+        if (!text) {
+            return;
+        }
+
+        this.setState({
+            loading: true,
+            error: false
+        });
+
+        const resourceClient = this.props.resourceClient || {
+            makeQuery: () => Promise.resolve([]),
+            parseResults: results => results
+        };
+
+        resourceClient.makeQuery(text)
+            .then(results => {
+                if (this.state.filterValue !== text) {
+                    this.setState({
+                        loading: false,
+                        error: false
+                    });
+
+                    return;
+                }
+                const itemGroups = [
+                    {
+                        items: resourceClient.parseResults(results)
+                    }
+                ];
+
+                this.setState({
+                    items: itemGroups,
+                    isOpen: true,
+                    loading: false,
+                    error: false
+                });
+            })
+            .catch(() => {
+                this.setState({
+                    error: true,
+                    loading: false,
+                    isOpen: false
+                });
+            });
+    }
+
+    /**
+     * Updates the selected items when a selection event occurs.
+     *
+     * @param {Object} item - The selected item.
+     * @private
+     * @returns {void}
+     */
+    _onSelectionChange(item) {
+        const existing
+            = this.state.selectedItems.find(k => k.value === item.value);
+        let selectedItems = this.state.selectedItems;
+
+        if (existing) {
+            selectedItems = selectedItems.filter(k => k !== existing);
+        } else {
+            selectedItems.push(item);
+        }
+        this.setState({
+            isOpen: false,
+            selectedItems
+        });
+
+        if (this.props.onSelectionChange) {
+            this.props.onSelectionChange(selectedItems);
+        }
+    }
+
+    /**
+     * Renders the content of this component.
+     *
+     * @returns {XML}
+     */
+    render() {
+        const shouldFitContainer = this.props.shouldFitContainer || false;
+        const shouldFocus = this.props.shouldFocus || false;
+        const isDisabled = this.props.isDisabled || false;
+        const placeholder = this.props.placeholder || '';
+        const noMatchesFound = this.props.noMatchesFound || '';
```

Comment:
I've been thinking that we should start using default props. Would they work here?